src/main/groovy/com/github/jrubygradle/internal/GemVersionResolver.groovy

```diff
@@ -15,14 +57,23 @@ class GemVersionResolver {
                     throw new RuntimeException( "there is no overlap for ${versions[details.requested.name]} and ${details.requested.version}" )
                 }
                 versions[details.requested.name] = next
+                log( { "${configuration.name}      collected ${version}" } )
+                log( { "${configuration.name}      resolved  ${next}" } )
                 details.useVersion next.toString()
             }
             else {
-                versions[details.requested.name] = new GemVersion(details.requested.version)
+                def next = new GemVersion(details.requested.version)
+                versions[details.requested.name] = next
+                log( { "${configuration.name}      nothing collected" } )
+                log( { "${configuration.name}      resolved  ${next}" } )
             }
         }
     }
 
+    void log(Closure message) {
```

Comment:
for testing - see my comment on no args constructor and you can probably check if the logger is doing info and avoid string concatenation here (well that is java thinking not sure about groovy)